File: algorithms/easy/minimum_cost_of_buying_candies_with_discount.ex
defmodule Solution do
  @spec minimum_cost(cost :: [integer]) :: integer
  def minimum_cost(cost) do
    cost
    |> Enum.sort(&>=/2)
    |> Enum.chunk_every(3)
    |> Enum.map(fn chunk -> chunk |> Enum.take(2) |> Enum.sum() end)
    |> Enum.sum()
  end
end

File: algorithms/easy/minimum_cost_of_buying_candies_with_discount.rs
impl Solution {
    pub fn minimum_cost(mut cost: Vec<i32>) -> i32 {
        cost.sort_by(|a, b| b.partial_cmp(a).unwrap());
        cost.chunks(3)
            .map(|chunk| chunk.into_iter().take(2).sum::<i32>())
            .sum()
    }
}
